main.js

```diff
@@ -1,173 +1,222 @@
 
-    // Make the request
-    req.send();
-  });
-}
+      req.open('GET', url);
+      req.responseType = responseType;
 
+      req.onload = () => {
+        const response = responseType ? req.response : req.responseXML;
 
+        if (response && response.errorMessages && response.errorMessages.length > 0){
+          reject(response.errorMessages[0]);
+          return;
+        }
 
-function loadOptions(){
-  chrome.storage.sync.get({
-    project: 'Sunshine',
-    user: 'nyx.linden'
-  }, function(items) {
-    document.getElementById('project').value = items.project;
-    document.getElementById('user').value = items.user;
-  });
-}
-function buildJQL(callback) {
-  var callbackBase = "https://jira.secondlife.com/rest/api/2/search?jql=";
-  var project = document.getElementById("project").value;
-  var status = document.getElementById("statusSelect").value;
-  var inStatusFor = document.getElementById("daysPast").value
-  var fullCallbackUrl = callbackBase;
-  fullCallbackUrl += 'project=${project}+and+status=${status}+and+status+changed+to+${status}+before+-${inStatusFor}d&fields=id,status,key,assignee,summary&maxresults=100';
-  callback(fullCallbackUrl);
-}
-function createHTMLElementResult(response){
-
-// 
-// Create HTML output to display the search results.
-// results.json in the "json_results" folder contains a sample of the API response
-// hint: you may run the application as well if you fix the bug. 
-// 
-
-  return '<p>There may be results, but you must read the response and display them.</p>';
-  
-}
-
-// utility 
-function domify(str){
-  var dom = (new DOMParser()).parseFromString('<!doctype html><body>' + str,'text/html');
-  return dom.body.textContent;
-}
-
-function checkProjectExists(){
+        resolve(response);
+      };
+
+      // Handle network errors
+      req.onerror = () => {
+        reject(Error('Network Error'));
+      }
+
+      req.onreadystatechange = () => { 
+        if(req.readyState === 4 && req.status === 401) { 
+          reject('You must be logged in to JIRA to see this project.');
+        }
+      }
+
+      // Make the request
+      req.send();
+    });
+  }
+
+  function loadOptions() {
+    chrome.storage.sync.get({
+      project: 'Sunshine',
+      user: 'nyx.linden',
+    }, (items) => {
+      document.getElementById('project').value = items.project;
+      document.getElementById('user').value = items.user;
+    });
+  }
+
+  function buildJQL(callback) {
+    const callbackBase = 'https://jira.secondlife.com/rest/api/2/search?jql=';
+    const project = document.getElementById('project').value;
+    const status = document.getElementById('statusSelect').value;
+    const inStatusFor = document.getElementById('daysPast').value
+
+    let fullCallbackUrl = callbackBase;
+
+    fullCallbackUrl += `project=${project}+and+status=${status}+and+status+changed+to+${status}+before+-${inStatusFor}d&fields=id,status,key,assignee,summary&maxresults=100`;
```

Comment:
This line will throw an error when submitting a query if you do not specify a period of query.